src/utils/wallet.js

```diff
@@ -132,32 +132,50 @@ export class Wallet {
       return await this.account.removeAccessKey(this.accountId, publicKey)
    }
 
-   async checkAccount(accountId) {
-      if (accountId !== this.accountId) {
-         return await this.near.nearClient.viewAccount(accountId)
-      } else {
-         throw new Error('You are logged into account ' + accountId + ' .')
+   async checkAccountAvailable(accountId) {
+      if (!this.isLegitAccountId(accountId)) {
+         throw new Error()
+      }
+      if (accountId === this.accountId) {
+         throw new Error()
+      }
+      let remoteAccount = null
+      try {
+         remoteAccount = await this.near.nearClient.viewAccount(accountId)
+      } catch (e) {
+         throw new Error()
```

Comment:
what. is the point of rethrowing new Error here? It does nothing except hiding actual reason of error.

Reply:
fixed as well